tests/usb/usb.go

```diff
@@ -0,0 +1,141 @@
+package usb
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	expect "github.com/google/goexpect"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/onsi/gomega/types"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	v1 "kubevirt.io/api/core/v1"
+	"kubevirt.io/api/usb/v1alpha1"
+	usbv1alpha1 "kubevirt.io/client-go/generated/kubevirt/clientset/versioned/typed/usb/v1alpha1"
+	"kubevirt.io/client-go/kubecli"
+
+	"kubevirt.io/kubevirt/tests"
+	"kubevirt.io/kubevirt/tests/console"
+	"kubevirt.io/kubevirt/tests/decorators"
+	"kubevirt.io/kubevirt/tests/framework/cleanup"
+	"kubevirt.io/kubevirt/tests/framework/kubevirt"
+	"kubevirt.io/kubevirt/tests/libvmi"
+	"kubevirt.io/kubevirt/tests/util"
+)
+
+var _ = Describe("[sig-compute-realtime][Serial]USB", Serial, decorators.SigCompute, func() {
+	var (
+		virtClient kubecli.KubevirtClient
+		usbClient  usbv1alpha1.UsbV1alpha1Interface
+	)
+
+	BeforeEach(func() {
+		virtClient = kubevirt.Client()
+		usbClient = virtClient.GeneratedKubeVirtClient().UsbV1alpha1()
+	})
+
+	Context("valid nodeconfig", func() {
+
+		const resourceName = "kubevirt.io/usb-storage"
+
+		createTestNodeConfig := func() *v1alpha1.NodeConfig {
+			nodeConfig := &v1alpha1.NodeConfig{
+				ObjectMeta: metav1.ObjectMeta{
+					GenerateName: "test",
+					Labels: map[string]string{
+						cleanup.TestLabelForNamespace(util.NamespaceTestDefault): "true",
+					},
+				},
+				Spec: v1alpha1.NodeConfigSpec{
+					// Validate there is at least one usb
+					USB: v1alpha1.USB{
+						ResourceName: resourceName,
+						USBHostDevices: []v1alpha1.USBHostDevices{
+							{
+								// Todo we could do VendorProductSelector and validate it
+								SelectByVendorProduct: "46f4:0001",
+							},
+						},
+					},
+				},
+			}
+			nodeConfig, err := usbClient.NodeConfigs().Create(context.TODO(), nodeConfig, metav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			return nodeConfig
+		}
+
+		It("sanity check", func() {
+			nodeConfig := createTestNodeConfig()
+
+			expectResourceToAppear(virtClient, resourceName)
+
+			By("Deleting node config")
+			err := usbClient.NodeConfigs().Delete(context.TODO(), nodeConfig.Name, metav1.DeleteOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			expectResourceToDisappear(virtClient, resourceName)
+		})
+
+		It("able to consume with VM", func() {
+			_ = tests.EnableFeatureGate("HostDevices")
```

Comment:
If you enable a feature gate you should run serial and then restore the old value. Maybe you should simply check if this is enabled and otherwise failing. Please, have a look how we do it for other hostdevices: https://github.com/kubevirt/kubevirt/blob/main/tests/vmi_hostdev_test.go#L39-L52

Additionally, please don't use the string but the constant `virtconfig.HostDevicesGate`

Reply:
Thanks!